ts/Core/Renderer/SVG/SVGElement.ts

```diff
@@ -1741,14 +1742,47 @@ class SVGElement implements SVGElementLike {
         eventType: string,
         handler: Function
     ): this {
-        const { onEvents } = this;
+        const element = this.element;
+        let touchStartPos: PositionObject;
+        let touchEventFired: boolean;
+
+        if (H.hasTouch && eventType === 'click') {
+            element.ontouchstart = function (e: TouchEvent): void {
+                // Store touch position
+                touchStartPos = {
+                    x: e.touches[0].clientX,
+                    y: e.touches[0].clientY
+                };
+            };
 
-        if (onEvents[eventType]) {
-            // Unbind existing event
-            onEvents[eventType]();
-        }
-        onEvents[eventType] = addEvent(this.element, eventType, handler);
+            // Call event on `ontouchend` instead of `ontouchstart` just like on
+            // `mouseup` event
+            element.ontouchend = function (e: TouchEvent): void {
+                // Check if chart was scrolled (touch position changed more than
+                // 4px)
+                const hasScrolled = touchStartPos.x ? Math.sqrt(
+                    Math.pow(touchStartPos.x - e.changedTouches[0].clientX, 2) +
+                    Math.pow(touchStartPos.y - e.changedTouches[0].clientY, 2)
+                ) >= 4 : false;
+
+                if (!hasScrolled) {
+                    handler.call(element, e);
+                }
+                touchEventFired = true;
 
+                if (e.cancelable !== false) {
+                    // Prevent other events from being fired. #9682
+                    e.preventDefault();
+                }
+            };
+            element.onclick = function (e: MouseEvent): void {
+                if (!touchEventFired) {
+                    handler.call(handler, e);
+                }
+            };
+        } else {
+            (element as any)['on' + eventType] = handler;
```

Comment:
I'm not happy about this `any` casting. 
@bre1470 Could you please have a look at this one?